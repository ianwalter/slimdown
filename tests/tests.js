const path = require('path')
const fs = require('fs')
const { test } = require('@ianwalter/bff')
const svelte = require('svelte/compiler')
const slimdown = require('..')

const filename = path.join(__dirname, 'fixtures/example.md')
const src = fs.readFileSync(filename)

test('preprocess', async ({ expect }) => {
  const context = {}
  const { code } = await svelte.preprocess(src, slimdown(context), { filename })
  expect(code).toMatchSnapshot()
  expect(context.frontmatter.slug).toBe('rich-man')
})
