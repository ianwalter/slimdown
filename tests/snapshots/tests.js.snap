// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`preprocess 1`] = `
"
<h1>Rich Man</h1>
"
`;
